account_invoice_import/wizard/account_invoice_import.py

```diff
@@ -0,0 +1,821 @@
+        #    'iso': 'EUR',
+        #    'currency_symbol': u'€',  # The one or the other
+        #    },
+        # 'date': '2015-10-08',  # Must be a string
+        # 'date_due': '2015-11-07',
+        # 'date_start': '2015-10-01',  # for services over a period of time
+        # 'date_end': '2015-10-31',
+        # 'amount_untaxed': 10.0,
+        # 'amount_tax': 2.0,  # provide amount_untaxed OR amount_tax
+        # 'amount_total': 12.0,  # Total with taxes, must always be provided
+        # 'partner': {
+        #       'vat': 'FR25499247138',
+        #       'email': 'support@browserstack.com',
+        #       'name': 'Capitaine Train',
+        #       },
+        # 'partner': res.partner recordset,
+        # 'invoice_number': 'I1501243',
+        # 'description': 'TGV Paris-Lyon',
+        # 'attachments': {'file1.pdf': base64data1, 'file2.pdf': base64data2},
+        # 'chatter_msg': ['Notes added in chatter of the invoice'],
+        # 'note': 'Note embedded in the document',
+        # 'origin': 'Origin note',
+        # 'lines': [{
+        #       'product': {
+        #           'barcode': '4123456000021',
+        #           'code': 'GZ250',
+        #           },
+        #       'name': 'Gelierzucker Extra 250g',
+        #       'price_unit': 1.45, # price_unit without taxes
+        #       'qty': 2.0,
+        #       'price_subtotal': 2.90,  # not required, but needed
+        #               to be able to generate adjustment lines when decimal
+        #               precision is not high enough in Odoo
+        #       'uom': {'unece_code': 'C62'},
+        #       'taxes': [{
+        #           'amount_type': 'percent',
+        #           'amount': 20.0,
+        #           'unece_type_code': 'VAT',
+        #           'unece_categ_code': 'S',
+        #           'unece_due_date_code': '432',
+        #           }],
+        #       'date_start': '2015-10-01',
+        #       'date_end': '2015-10-31',
+        #       # date_start and date_end on lines override the global value
+        #       }],
+        # }
+
+        # IMPORT CONFIG
+        # {
+        # 'invoice_line_method': '1line_no_product',
+        # 'account_analytic': Analytic account recordset,
+        # 'account': Account recordset,
+        # 'taxes': taxes multi-recordset,
+        # 'label': 'Force invoice line description',
+        # 'product': product recordset,
+        # }
+        #
+        # Note: we also support importing customer invoices via
+        # create_invoice() but only with 'nline_*' invoice import methods.
+
+    @api.model
+    def _prepare_create_invoice_vals(self, parsed_inv, import_config=False):
+        assert parsed_inv.get('pre-processed'), 'pre-processing not done'
+        aio = self.env['account.invoice']
+        ailo = self.env['account.invoice.line']
+        bdio = self.env['business.document.import']
+        rpo = self.env['res.partner']
+        company = self.env.user.company_id
```

Comment:
You should import the company from the invoice and leave this way when company cannot be found.

Reply:
Don't understand: we do not have company in imported invoice .. If you want to import some invoice for one special company, juste change company before importing ?

Reply:
The only way I see is to look for the partner company, and if not present, then use this default.

Reply:
The user could have selected the wrong company. If the system detects the real company, it simplifies all the process and it removes possible problems.
Moreover, if you want to integrate the invoices using some standard format, you don't want to change the company of the user each time.
@pedrobaeza That is what I thought.

Reply:
You could do
company = self._get_company(parsed_inv, import_config)
def _get_company(self, parsed_inv, import_config):
    return self.env.user.company_id

Then, if someone wants to parse the company its easy as we have a hook.

Reply:
Ok will do this :)